### PR TITLE
#4913 Call `updateInitialCartData` in router on every history change

### PR DESCRIPTION
**Pull request by neeeeecka**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/4913

**Problem:**
* Since we have an SPA and don't really have any mechanisms to get realtime/newest data from server yet, the cart data was fetched only on router's initialization and few other times.

**In this PR:**
* I modified `Router.container.js` to listen to route changes through `history.listen`, now its fetching new cart data every time route changes(Something like Luma), that gives a bit fresher data to user.

### CODE REVIEW

File: packages/scandipwa/src/component/Router/Router.container.js [hunk 1]
```diff
@@ -137,6 +148,10 @@ export class RouterContainer extends PureComponent {
 
     componentDidMount() {
         window.addEventListener('resize', this.handleResize);
+
+        history.listen(this.handleRouteChange([
+            this.updateCartOnRouteChange
+        ]));
```
Comment (zans-laksa) [inline packages/scandipwa/src/component/Router/Router.container.js:154]: But there is just one callback and unlikely we will add another one.
Please reduce the logic here. It's too complex.

**Review by zans-laksa — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Router/Router.container.js (hunk 1 above)
Comment (neeeeecka) [reply packages/scandipwa/src/component/Router/Router.container.js:154]: https://github.com/scandipwa/scandipwa/issues/4971

I will push this in a few minutes :D

**Comment by aleksandrakorolova** [comment]
1. Product price doesn't update in the cart/mini-cart if user changed it in Admin panel
**Notes:** It's updated if user changes qty of product or deletes, adds product again 
Screenshot 2022-07-15 at 10 51 48

2. Qty stays as user added but only + button becomes disabled in the cart/mini-cart if user changed Max Qty Allowed in Shopping Cart in Admin panel
Screenshot 2022-07-15 at 11 19 26
**Steps:
FE:**
-Open product X
-Change qty to 3
-Add product to cart
**Admin panel:**
-Open Catalog -> Product 
-Open Product X
-Open Advanced Inventory
-Change Max Qty Allowed in Shopping Cart to 1
-Save
**FE:**
-Return back to FE
-Open Cart

**Expected result:**
Product qty changes to Max Qty Allowed in Shopping Cart and + button is disabled
**Notes:**
Minimum Qty Allowed in Shopping Cart works good. The same as name, stock status

**Review by zans-laksa — APPROVED** [review]

**Comment by aleksandrakorolova** [comment]
1. Can't reproduce
2. Can't test because of another issue - #5153

**Comment by AleksandrsKondratjevs** [comment]
Closing due to depricated code.

**Pull request closed without merging**